app/models/comment.rb

```diff
@@ -282,34 +287,13 @@ def send_email_notification
   end
 
   def synchronous_spam_score_check
-    return unless
-      Settings::RateLimit.spam_trigger_terms.any? { |term| Regexp.new(term.downcase).match?(title.downcase) }
+    return unless Settings::RateLimit.trigger_spam_for?(text: [title, body_markdown].join("\n"))
 
     self.score = -1 # ensure notification is not sent if possibly spammy
   end
 
   def create_conditional_autovomits
-    return unless
-      Settings::RateLimit.spam_trigger_terms.any? { |term| Regexp.new(term.downcase).match?(title.downcase) } &&
-        user.registered_at > 5.days.ago
```

Comment:
I'm uncertain what this bit of logic was intending.  We have the following four:

1. TRUE - One of the terms was spam AND the user registered more than 5 days ago.
2. FALSE - One of the terms was spam AND the user registered less than five days ago.
3. FALSE - No spam terms AND the user registered more than 5 days ago.
4. FALSE - No spam terms AND the user registered less than five days ago

We return `nil` in list items 2, 3, and 4.  So we treat something as spam if the term has spam but it's a not so new user.

Reply:
@benhalpern I did some annotating, and you introduced this logic.  Could you help me understand the intention?

https://github.com/forem/forem/blame/ca0db28fc425b38b5fa2c20afe7e1731eec915f7/app/models/comment.rb#L294

Reply:
In conversations with Success team members, spammers are likely to start right away